tests/pytest/test_pytest_runner.py

```diff
@@ -0,0 +1,110 @@
+    mocker.patch("app_test_suite.steps.base_test_runner.ChartMuseumAppRepository.upload_artifacts")
+    mocker.patch("app_test_suite.steps.base_test_runner.create_app")
+    mocker.patch("app_test_suite.steps.base_test_runner.wait_for_apps_to_run")
+    mocker.patch("app_test_suite.steps.base_test_runner.delete_app")
+    mocker.patch("app_test_suite.steps.base_test_runner.wait_for_app_to_be_deleted")
+
+    mocker.patch("app_test_suite.steps.pytest.pytest.run_and_log", return_value=system_call_result_mock)
+
+    config = mocker.Mock(name="ConfigMock")
+    config.app_tests_skip_app_deploy = False
+    config.app_tests_deploy_namespace = mock_app_deploy_ns
+    config.app_tests_app_config_file = ""
+    config.chart_file = "mock_chart.tar.gz"
+    context = {context_key_chart_yaml: {"name": mock_app_name, "version": mock_app_version}}
+    runner = PytestTestTimeRunner(mock_cluster_manager)
+    runner.run(config, context)
+
+    # assert step created connection with correct config file
+    cast(unittest.mock.Mock, pykube.KubeConfig.from_file).assert_called_once_with(mock_kube_config_path)
+    cast(unittest.mock.Mock, app_test_suite.steps.base_test_runner.HTTPClient).called_once()
+    # assert ensure app platform ready
+    cast(unittest.mock.Mock, app_test_suite.steps.base_test_runner.run_and_log).assert_called_with(
+        ["apptestctl", "bootstrap", f"--kubeconfig-path={mock_kube_config_path}", "--wait"]
+    )
+    # uploaded the correct chart to chart repository
+    cast(
+        unittest.mock.Mock, app_test_suite.steps.base_test_runner.ChartMuseumAppRepository.upload_artifacts
+    ).assert_called_once_with(config, context)
+    # deploys app cr and waits for it to run
+    cast(unittest.mock.Mock, app_test_suite.steps.base_test_runner.create_app).assert_called_once_with(
+        unittest.mock.ANY, mock_app_name, mock_app_version, "chartmuseum", "default", mock_app_deploy_ns, None
+    )
+    cast(unittest.mock.Mock, app_test_suite.steps.base_test_runner.wait_for_apps_to_run).assert_called_once_with(
+        unittest.mock.ANY, [mock_app_name], "default", BaseTestRunner._app_deployment_timeout_sec
+    )
+    # installs deps with pipenv and runs pytest
+    assert cast(unittest.mock.Mock, app_test_suite.steps.pytest.pytest.run_and_log).call_args_list[0].args[0] == [
+        "pipenv",
+        "install",
+        "--deploy",
+    ]
+    cast(unittest.mock.Mock, app_test_suite.steps.pytest.pytest.run_and_log).assert_called_with(
+        [
+            "pipenv",
+            "run",
+            "pytest",
+            "-m",
+            STEP_TEST_TEST,
+            "--cluster-type",
+            "mock",
+            "--kube-config",
+            mock_kube_config_path,
+            "--chart-path",
+            config.chart_file,
+            "--chart-version",
+            mock_app_version,
+            "--chart-extra-info",
+            "external_cluster_version=1.19.1",
```

Comment:
`1.19.1` is the kind version?

What about having a constant for this?  So it's easier to find and update.

Also 1.19.1 is quite old now. For the Go tests we use 1.21.1 and the latest kind tag is 1.21.2. I think we should use at least K8s 1.20.2.

https://quay.io/repository/giantswarm/kind-node

Reply:
It's total mock, doesn't have anything to do with real version used anywhere. Real cluster manager is not used here, only the mocked one, so this value is never used anywhere :) Maybe I should set it to something ridiculous, so it's obvious :)

Reply:
OK I missed that completely. 😹  What about `2.0.0`? :trollface:  

I think it's fine maybe just bump it to 1.21. It was the 1.19 that triggered me.

Reply:
OK its fine as it. No need to waste more time on this.